src/control/server/storage/bdev/backend_json.go

```diff
@@ -237,5 +237,24 @@ func newSpdkConfig(log logging.Logger, req *storage.BdevWriteConfigRequest) (*Sp
 		}
 	}
 
+	if req.HotplugEnabled {
+		for _, ss := range sc.Subsystems {
+			if ss.Name != "bdev" {
+				continue
+			}
+
+			for _, bsc := range ss.Configs {
+				if bsc.Method == SpdkBdevNvmeSetHotplug {
+					bsc.Params = NvmeSetHotplugParams{
+						Enable: true,
+						PeriodUsec: 10 * 1000 * 1000,
```

Comment:
Instead of setting an arbitrary value here, I recommend adding a const at the top of the file and using it here, e.g.
```go
const (
        HotplugPeriod = uint64(10 * time.Second)
)
```

In Go, [time.Duration](https://pkg.go.dev/time#Duration) is a type alias for int64, so you can define durations with human-readable values but use them in the same place as a int64. In this case we just have to cast it to uint64 because that's what the struct field needs.

Reply:
Sure, but I don't quite see what's the 'time.Second'? Maybe make it "HotplugPeriod = 5 * 1000 * 1000? 10 seconds hotplug period (that's the maximum hotplug polling period) looks to long to me.

Reply:
[time.Second](https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/time/time.go;l=612) is one of several const values defined for making durations easier for humans to understand. So, 10 * 1000 * 1000 (i.e. 10,000,000) microseconds is the same as 10 seconds, right?. The math is not difficult, but it's not clear to the reader. Much easier to just read `10 * time.Second` and say "oh, 10 seconds. that's easy".

Reply:
In Go, [time.Second](https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/time/time.go;l=612) is one of several const values defined to make it easier for humans to work with durations.

What is the desired hotplug period? The field `PeriodUsec` wants a value in microseconds, I assume from the name. So you have `10 * 1000 * 1000` which is 10,000,000, and 10,000,000 microseconds is 10s, right?

In my suggestion I forgot to convert the const to microseconds, so it would be best to define the value in nanoseconds (remember `time.Second` is a const defined as 1,000,000,000ns), and then convert it as necessary where it's consumed, e.g.

```go
const (
       HotPlugPeriod = 10 * time.Second (or 10 * time.Millisecond if that's what you meant)
)
```

... then convert it where it's consumed

```go
bsc.Params = NvmeSetHotplugParams{
        PeriodUsec: uint64(HotplugPeriod / time.Microsecond), // get the value in us
}
```

Not trying to make it more complicated, just trying to make it easier for other people to read and maintain over the years. This is how we do things in the rest of the Go code.

Reply:
I see, thanks a lot for the explanation! I'll change to use time.Second if the patch needs be refreshed, otherwise, I'll make the patch in follow-on hotplug filter patch.